src/simd.rs

```diff
@@ -383,6 +433,42 @@ mod test {
             .boxed()
     }
 
+    #[derive(Debug)]
+    struct Ranges {
+        ranges: [(u8, u8); 8],
+        len: usize,
+    }
+
+    impl Ranges {
+        fn as_slice(&self) -> &[(u8, u8)] {
+            &self.ranges[..self.len]
+        }
+    }
+
+    /// Creates an array of valid ranges (where the first number is less-than-or-equal-to
+    /// the second number) and the number of such ranges.
+    fn valid_ranges() -> BoxedStrategy<Ranges> {
+        (any::<[(u8, u8); 8]>(), 0..=8_usize)
+            .prop_map(|(pairs, len)| {
+                let mut ranges: [(u8, u8); 8] = pairs;
+                for (ref lo, hi) in ranges.iter_mut() {
+                    let extra = *hi;
+                    *hi = if extra > 255_u8 - lo { 255_u8 } else { lo + extra };
```

Comment:
this might be clearer as either of these pseudocode:

1. ```rust
    let mut x = [a, b];
    x.sort();
    let [a, b] = x;
    ```

2. `any::<u8>.prop_map(|a| (a, range(a, u8::max)))`

Reply:
or `if a < b { mem::swap(a, b) }`